src/test/java/io/confluent/kafkarest/integration/ClusterTestHarness.java

```diff
@@ -51,19 +57,45 @@
 
   public static final int DEFAULT_NUM_BROKERS = 1;
 
-  // Shared config
-  protected Queue<Integer> ports;
+  /**
+   * Choose a number of random available ports
+   */
+  public static int[] choosePorts(int count) {
+    try {
+      ServerSocket[] sockets = new ServerSocket[count];
+      int[] ports = new int[count];
+      for (int i = 0; i < count; i++) {
+        sockets[i] = new ServerSocket(0);
+        ports[i] = sockets[i].getLocalPort();
+      }
+      for (int i = 0; i < count; i++)
+        sockets[i].close();
+      return ports;
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  /**
+   * Choose an available port
+   */
+  public static int choosePort() {
```

Comment:
Is there any danger that successive calls to `choosePort` will return the same port?

It looks like before the change you allocated all ports at once, and then handed them out from that pool.